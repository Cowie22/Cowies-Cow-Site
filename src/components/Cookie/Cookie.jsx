import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import AppProvider, { AppContext } from '../../contexts/state';

import ExternalLink from "../exitramp/ExternalLink.jsx"

import closeIcon from '../../assets/images/close.svg';
import closeIconHovered from '../../assets/images/close-hovered.svg';

class Cookie extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hovered: false,
    }
  }

  handleHovered = (val) => {
    this.setState({
      hovered: val,
    })
  }

  render() {
    const { hovered } = this.state;
    return (
      <AppContext.Consumer>
        {(context) => {
          const { isCookieVisible, handleIsCookieVisible } = context;
          return (
            isCookieVisible ?
            <section className='cookie-container' data-testid='Cookie Visible'>
              <Container>
                <Row>
                  <Col lg={9} xs={{span: 8}}>
                    <p>
                      This website uses cookies to help us give you the best experience when you visit.
                      By using this website, you consent to the use of these cookies. Find out more about
                      how we use cookies and how to manage them by reading our
                      <ExternalLink
                        externalUrl={'https://www.myovant.com/cookie-policy/'}
                        className='cookie-link extra-bold'
                      >
                        {' '} Cookie Policy
                      </ExternalLink>.
                    </p>
                  </Col>
                  <Col lg={3} xs={{span: 4}}>
                    <div className='cookie-btn-container'>
                      <Row>
                        <Col lg={{span: 9, order: 1}} md={{order: 1}} xs={{order: 2}}>
                          <div className='cta-btn-container cookie-cta-container' onClick={() => handleIsCookieVisible(false)}>
                            <button className='cta-btn transparent-btn-pink cookie-cta-btn'>
                              Accept
                            </button>
                          </div>
                        </Col>
                        <Col lg={{span: 3, order: 2}}  md={{order: 2}} xs={{order: 1}}>
                          <div
                            className='close-icon-container'
                            onMouseEnter={() => this.handleHovered(true)}
                            onMouseLeave={() => this.handleHovered(false)}
                          >
                            <img
                              loading='lazy'
                              src={hovered ? closeIconHovered : closeIcon}
                              onClick={() => handleIsCookieVisible(false)}
                              className='header-close-icon'
                              alt=''
                            />
                          </div>
                        </Col>
                      </Row>
                    </div>
                  </Col>
                </Row>
              </Container>
            </section>
            :
            <div data-testid='Cookie Hidden'>

            </div>
          )
        }}
      </AppContext.Consumer>
    )
  }
}

export default Cookie