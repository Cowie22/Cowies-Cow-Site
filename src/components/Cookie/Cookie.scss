.cookie-container {
  padding: 16px 0;
  .cookie-link {
    text-decoration: none;
    color: $purple;
    &:hover {
      cursor: pointer;
    }
  }
  .close-icon-container {
    display: flex;
    align-items: flex-end;
    justify-content: flex-end;
    .header-close-icon {
      &:hover {
        cursor: pointer;
      }
    }
  }
}