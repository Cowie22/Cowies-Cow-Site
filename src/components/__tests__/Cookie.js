import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import Cookie from '../Cookie/Cookie'

const { debug, getByAltText, getByText, getByTestId, getAllByText, container } = render( <Cookie /> )

const cookiePolicy = getByText('Cookie Policy')
const acceptBTN = getByText('Accept')
const cookieVisible = getByTestId('Cookie Visible')

test('Cookie links exist', () => {
  // debug()
  expect(cookiePolicy).toBeTruthy()
  expect(cookieVisible).toBeTruthy()
})

test('Cookie links have appropriate href', () => {
  expect(cookiePolicy.getAttribute('href')).toBe('https://www.myovant.com/cookie-policy/')
})

test('Cookie accept link closes cookie statement', async () => {
	const clickBTN = () => {
		fireEvent.click(acceptBTN.parentElement)
	}

	await waitFor(() => clickBTN())

	setTimeout(() => {
		expect(getByTestId('Cookie Hidden')).toBeTruthy()
	}, 2000)
})

test('Cookie snapshot matches', () => {
//   return
  // once the component is done, use snapshot
  expect(container.firstChild).toMatchSnapshot()
})