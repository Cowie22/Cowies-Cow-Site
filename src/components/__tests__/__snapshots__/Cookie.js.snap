// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie snapshot matches 1`] = `null`;
